components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -1786,12 +1468,556 @@ subroutine ma_convproc_tend(                                           &
+       endif
+       wup(kk) = min_max_bound(0.1_r8, w_peak, wup(kk))
+   endif
+
+   end subroutine compute_wup
+
+!====================================================================================
+   subroutine compute_activation_tend(                          &
+                        icol,   kk,     clw_cut,        f_ent,  & ! in
+                        cldfrac_i,      rhoair_i,               & ! in
+                        dt_u,   wup,    icwmr,          t,      & ! in
+                        kactcnt,        kactfirst,              & ! inout
+                        conu,           dconudt_activa          ) ! inout
+!-----------------------------------------------------------------------
+! aerosol activation - method 2
+!    when kactcnt=1 (first/lowest layer with cloud water)
+!       apply "primary" activatation to the entire updraft
+!    when kactcnt>1
+!       apply secondary activatation to the entire updraft
+!       do this for all levels above cloud base (even if completely glaciated)
+!          (this is something for sensitivity testing)
+!-----------------------------------------------------------------------
+   use ppgrid, only: pver, pverp, pcols
+   use constituents, only: pcnst
+
+   ! cloudborne aerosol, so the arrays are dimensioned with pcnst_extd = pcnst*2
+   integer,  parameter  :: pcnst_extd = pcnst*2
+   integer,  intent(in) :: icol                 ! column index
+   integer,  intent(in) :: kk                   ! vertical level index
+   real(r8), intent(in) :: clw_cut              ! threshold clw value for doing updraft [kg/kg]
+   real(r8), intent(in) :: cldfrac_i(pver)      ! cldfrac at current icol (with adjustments) [fraction]
+   real(r8), intent(in) :: rhoair_i(pver)       ! air density at current i [kg/m3]
+   real(r8), intent(in) :: f_ent                ! fraction of updraft massflux that was entrained across this layer == eudp/mu_p_eudp [fraction]
+   real(r8), intent(in) :: dt_u(pver)           ! lagrangian transport time in the updraft at current level [s]
+   real(r8), intent(in) :: wup(pver)            ! mean updraft vertical velocity at current level updraft [m/s]
+   real(r8), intent(in) :: t(pcols,pver)        ! Temperature [K]
+   real(r8), intent(in) :: icwmr(pcols,pver)    ! Convective cloud water from zm scheme [kg/kg]
+   integer,  intent(inout) :: kactcnt           !  Counter for no. of levels having activation
+   integer,  intent(inout) :: kactfirst         ! Lowest layer with activation (= cloudbase)
+   real(r8), intent(inout) :: conu(pcnst_extd,pverp)   ! mix ratio in updraft at interfaces [kg/kg]
+   real(r8), intent(inout) :: dconudt_activa(pcnst_extd,pverp) ! d(conu)/dt by activation [kg/kg/s]
+
+   logical      :: do_act_this_lev      ! flag for doing activation at current level
+   integer      :: kp1                  ! kk + 1
+
+   ! initiate variables
+   do_act_this_lev = .false.
+   kp1 = kk+1
+
+   if (kactcnt <= 0) then
+       if (icwmr(icol,kk) > clw_cut) then
```

Comment:
Instead of passing `clw_cut`, we can declare it as a parameter at the top of this module. If the variable is a parameter, it is perfectly fine to declare it at the top and use it.

Reply:
will do in this PR